code/modules/hydroponics/compost_bin.dm

```diff
@@ -0,0 +1,171 @@
+#define BASE_BIOMASS_CAPACITY 1500
+#define BASE_COMPOST_CAPACITY 1500
+#define SOIL_COST 25
+#define DECAY 0.2
+#define MIN_CONVERSION 10
+/**
+  * # compost bin
+  * used to make soil from plants.
+  * Doesn't have components.
+  */
+/obj/machinery/compost_bin
+	name = "compost bin"
+	desc = "A wooden bin for composting."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "compost_bin-empty"
+	power_state = NO_POWER_USE
+	density = TRUE
+	anchored = TRUE
+	/// stage in the process, for timing purposes.
+	var/process_counter = 0
+	/// amount of biomass in the compost bin
+	var/biomass = 0
+	/// amount of compost in the compost bin
+	var/compost = 0
+	/// The maximum amount of biomass the compost bin can store.
+	var/biomass_capacity = BASE_BIOMASS_CAPACITY
+	/// The maximum amount of compost the compost bin can store.
+	var/compost_capacity = BASE_COMPOST_CAPACITY
```

Comment:
why do you use a define for these instead of just putting the number on the var declaration?

Reply:
I want to have a base amount set, and I figured a define would be best here. I know the vars are self explanatory, but it seemed odd to just shove a number in there.

Reply:
Just trying to avoid "magic numbers", really.

Reply:
Defines are usually used when you use the number multiple times across different procs and it can't change, using defines in the matter you're using them here makes it more confusing if someone were to add a subtype later. You also just use the variable in all its references instead of the define, so replacing the define with the numbers wouldn't really be a "magic number"